presto-main/src/main/java/com/facebook/presto/server/remotetask/ContinuousTaskStatusFetcher.java

```diff
@@ -0,0 +1,242 @@
+        if (!running || taskStatus.getState().isDone()) {
+            return;
+        }
+
+        // outstanding request?
+        if (future != null && !future.isDone()) {
+            // this should never happen
+            log.error("Can not reschedule update because an update is already running");
+            return;
+        }
+
+        // if throttled due to error, asynchronously wait for timeout and try again
+        ListenableFuture<?> errorRateLimit = errorTracker.acquireRequestPermit();
+        if (!errorRateLimit.isDone()) {
+            errorRateLimit.addListener(this::scheduleNextRequest, executor);
+            return;
+        }
+
+        Request request = prepareGet()
+                .setUri(uriBuilderFrom(taskStatus.getSelf()).appendPath("/status").build())
+                .setHeader(CONTENT_TYPE, JSON_UTF_8.toString())
+                .setHeader(PRESTO_CURRENT_STATE, taskStatus.getState().toString())
+                .setHeader(PRESTO_MAX_WAIT, refreshMaxWait.toString())
+                .build();
+
+        errorTracker.startRequest();
+        future = httpClient.executeAsync(request, createFullJsonResponseHandler(taskStatusCodec));
+        Futures.addCallback(future, new SimpleHttpResponseHandler<>(this, request.getUri()), executor);
+    }
+
+    TaskStatus getTaskStatus()
+    {
+        return taskStatus.get();
+    }
+
+    @Override
+    public void success(TaskStatus value)
+    {
+        try (SetThreadName ignored = new SetThreadName("ContinuousTaskInfoFetcher-%s", taskId)) {
+            try {
+                updateTaskStatus(value);
+                errorTracker.requestSucceeded();
+            }
+            finally {
+                scheduleNextRequest();
+            }
+        }
+    }
+
+    @Override
+    public void failed(Throwable cause)
+    {
+        try (SetThreadName ignored = new SetThreadName("ContinuousTaskInfoFetcher-%s", taskId)) {
+            try {
+                // if task not already done, record error
+                TaskStatus taskStatus = getTaskStatus();
+                if (!taskStatus.getState().isDone()) {
+                    errorTracker.requestFailed(cause);
+                }
+            }
+            catch (Error e) {
+                task.failTask(e);
+                throw e;
+            }
+            catch (RuntimeException e) {
+                task.failTask(e);
+            }
+            finally {
+                // there is no back off here so we can get a lot of error messages when a server spins
+                // down, but it typically goes away quickly because the queries get canceled
+                scheduleNextRequest();
+            }
+        }
+    }
+
+    @Override
+    public void fatal(Throwable cause)
+    {
+        try (SetThreadName ignored = new SetThreadName("ContinuousTaskInfoFetcher-%s", taskId)) {
+            task.failTask(cause);
+        }
+    }
+
+    synchronized void updateTaskStatus(TaskStatus newValue)
```

Comment:
When we send an update to the remote task, we return a taskInfo object. I tried ignoring the return value and running taskStatus updates as a fixed cadence, but noticed that the abstract test queries were consistently 30% slower. So I added method which is called from the RemoteTask.